src/PasswordInputStateful/index.jsx

```diff
@@ -0,0 +1,254 @@
+
+import { PasswordInput }    from '../index';
+
+export default class PasswordInputStateful extends React.Component
+{
+    static propTypes =
+    {
+        /**
+         *  ARIA properties
+         */
+        aria : PropTypes.objectOf( PropTypes.oneOfType( [
+            PropTypes.bool,
+            PropTypes.number,
+            PropTypes.string,
+        ] ) ),
+        /**
+         *  Extra CSS class name
+         */
+        className             : PropTypes.string,
+        /**
+         *  CSS class map
+         */
+        cssMap                : PropTypes.objectOf( PropTypes.string ),
+        /**
+         *  Tooltip message text (string or JSX)
+         */
+        errorMessage          : PropTypes.node,
+        /**
+         *  Error Tooltip is displayed
+         */
+        errorMessageIsVisible : PropTypes.bool,
+        /**
+        *   Error message position relative to the icon
+        */
+        errorMessagePosition  : PropTypes.oneOf( [ 'top', 'topLeft' ] ),
+        /**
+         *  Display as hover when required from another component
+         */
+        forceHover            : PropTypes.bool,
+        /**
+         *  Display as error/invalid
+         */
+        hasError              : PropTypes.bool,
+        /**
+         *  Display Button icon as disabled
+         */
+        iconButtonIsDisabled  : PropTypes.bool,
+        /**
+         *  Alignment of the icon
+         */
+        iconPosition          : PropTypes.oneOf( [ 'left', 'right' ] ),
+        /**
+         *  Display the icon tooltip
+         */
+        iconTooltipIsVisible  : PropTypes.bool,
+        /**
+         *  icon Tooltip message text (string or JSX)
+         */
+        iconTooltipMessage    : PropTypes.node,
+        /**
+         *  Icon Tooltip position relative to icon
+         */
+        iconTooltipPosition   : PropTypes.oneOf( [
+            'left',
+            'right',
+            'top',
+            'bottom',
+            'topLeft',
+            'topRight',
+        ] ),
+        /**
+         *  HTML id attribute
+         */
+        id               : PropTypes.string,
+        /**
+         *  Callback that receives the native <input>: ( ref ) => { ... }
+         */
+        inputRef         : PropTypes.func,
+        /**
+         *  Display as disabled
+         */
+        isDisabled       : PropTypes.bool,
+        /**
+         *  Display as read-only
+         */
+        isReadOnly       : PropTypes.bool,
+        /**
+         *  Display as read-only for IconButton
+         */
+        isReadOnlyButton : PropTypes.bool,
+        /**
+         *  Display as read-only for TextInput
+         */
+        isReadOnlyInput  : PropTypes.bool,
+        /**
+         *  Label text (string or JSX node)
+         */
+        label            : PropTypes.node,
+        /**
+         *  Label position
+         */
+        labelPosition    : PropTypes.oneOf( [ 'top', 'left', 'right' ] ),
```

Comment:
`inputContainer` component has been removed, so there is no need for label related props.